composer/utils/checkpoint.py

```diff
@@ -510,34 +573,33 @@ def read_data(self, plan: LoadPlan, planner: LoadPlanner):
                 # Ensure state exists
                 state_dict['state'] = state_dict.get('state', {})
 
-            # Only some ranks are meant to load checkpoint
-            expect_file = False
+            # Get shard process group for HSDP
             process_group = None
+            coordinator_rank = 0
             device_mesh = state.fsdp_device_mesh
             if device_mesh is not None and device_mesh.ndim == 2:
-                # If hybrid shard, only rank in first replica saves
-                expect_file = (device_mesh.get_local_rank(mesh_dim=0) == 0)
-                if expect_file:
-                    process_group = device_mesh.get_group(1)  # Shard process_group for first replica
-                    log.debug(f'global_rank={dist.get_global_rank()}, {expect_file=}')
-            else:
-                expect_file = True
+                process_group = device_mesh.get_group(1)
+                shard_size = device_mesh.size(1)
+                coordinator_rank = dist.get_global_rank() % shard_size
 
             if version.parse(torch.__version__) > version.parse('2.2.9'):
                 dist_cp.load(  # type: ignore
                     state_dict=state_dict,
                     storage_reader=storage_reader,
                     planner=load_planner,
                     process_group=process_group,
```

Comment:
your PR description says you removed the pg from load, but i see it here? could you explain more